src/pal/src/file/directory.cpp

```diff
@@ -609,13 +600,6 @@ CreateDirectoryA(
     // Canonicalize the path so we can determine its length.
     FILECanonicalizePath(realPath);
```

Comment:
Now that we don't check path length any more, there is no need to canonicalize the path.